packages/fx-core/src/plugins/resource/frontend/utils.ts

```diff
@@ -21,6 +22,10 @@ export class Utils {
     return raw.replace(RegularExpr.AllCharToBeSkippedInName, Constants.EmptyString).toLowerCase();
   }
 
+  static capitalizeFirstLetter([first, ...rest]: Iterable<string>): string {
+    return [first?.toUpperCase(), ...rest].join("");
```

Comment:
Nit: what if string is empty? I am not familiar with this style. Want a test result for it.

Reply:
If string is empty, the `first` element would be undefined and the method return empty.
```
> (([first, ...rest]) => {
      return {first: first, rest: rest, value: [first?.toUpperCase(), ...rest].join("")};
  }) ("aaaa")
{first: "a", rest: Array(3), value: "Aaaa"}
> (([first, ...rest]) => {
      return {first: first, rest: rest, value: [first?.toUpperCase(), ...rest].join("")};
  }) ("")
{first: undefined, rest: Array(0), value: ""}
```

Reply:
cool

Reply:
The array indexed solution can only deal with UTF-16 string, although I think it's enough for path.
I just think the destructuring assignment looks fancier.